src/repositories/user.repository.ts

```diff
@@ -0,0 +1,35 @@
+import { Connection, Repository } from "typeorm";
+
+import { CreateUserDto, UpdateUserDto } from "../dtos/user.dto";
+import { UserModel } from "../models/user.model";
+
+export class UserRepository {
+  private repository: Repository<UserModel>;
+
+  constructor(connection: Connection) {
+    this.repository = connection.getRepository(UserModel);
+  }
+
+  public async create(payload: CreateUserDto) {
+    const user = this.repository.create(payload);
```

Comment:
I think an await might be missing here

Reply:
Sorry for the late response, this method is not asynchronous. 

The async method is the one below, when the user is saved to the DB. This video explains it better, I guess, it is about the repository pattern in TS 😄: https://www.youtube.com/watch?v=XU0w62XQjpA&t=794s&ab_channel=DonnyRoufs